src/jdk.jdwp.agent/share/native/libjdwp/threadControl.c

```diff
@@ -2179,6 +2180,59 @@ threadControl_onEventHandlerEntry(jbyte sessionID, EventInfo *evinfo, jobject cu
 static void
 doPendingTasks(JNIEnv *env, ThreadNode *node)
 {
+    /* Deferred breakpoint handling for Thread.resume() */
+    if (node->handlingAppResume) {
+        jthread resumer = node->thread;
+        jthread resumee = getResumee(resumer);
+
+        /*
+         * trackAppResume() (indirectly) aquires handlerLock. For proper lock
+         * ordering handlerLock has to be acquired before threadLock.
+         */
+        debugMonitorExit(threadLock);
+        eventHandler_lock();
+        debugMonitorEnter(threadLock);
```

Comment:
The lines 2192-2194 look like a hack to me.
But I see you have filed an enhancement to fix this.
I'm okay to address it later.

Reply:
Thanks for looking at this Serguei.
I agree that this part looks a little odd. It is still correct, though, and the changes are limited to the hardly used code responsible for handling j.l.Thread.resume() calls which is a good thing IMHO.